### DISCUSSION TITLE
Help for autodiff()

### DISCUSSION THREAD

**Discussion by alopezs3** [lgrcia/prose#154] [Q&A]
Hello, Mr. García and the community. First, I want to thank you for this powerful package. I am new to PROSE, so I went through the photometry tutorial. In particular, I am working on differential photometry for my thesis. However, I had already performed aperture photometry separately. My goal is to use PROSE for differential photometry and obtain light curves. In my case, I have bidimensional arrays of flux and error (for a single aperture), along with time. So, what I did was create an object of the Fluxes class from prose.fluxes, using the arrays I mentioned before. It indeed created the object of the class successfully. The issue arose when I tried to use the autodiff() function, as it throws the following error:

```
File ~/anaconda3/lib/python3.11/site-packages/prose/fluxes.py:71, in weights(fluxes, tolerance, max_iteration, bins)
     67     lcs = diff(dfluxes, weights=weights)
     69     i += 1
---> 71 weights[0, mask] = 0
     73 return weights[0]

IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
```

I would greatly appreciate any help to resolve the error. Adjoint my code:

```python
from prose.fluxes import Fluxes

#flux2 es un ndarray de dimensión 2 al igual que errors, time es de dimensión 1
raw_fluxes = Fluxes(fluxes=flux2,time=time,errors=errors)

raw_fluxes.target = 1000

nan_stars = np.any(np.isnan(raw_fluxes.fluxes), axis=(0, 2)) # stars with nan fluxes

fluxes = raw_fluxes.mask_stars(~nan_stars)

diff_fluxes = raw_fluxes.autodiff()
```

**Comment by lgrcia** [comment]
Hi @alopezs3 and thanks for your question!

Just to be sure we are on the same page, here is a simulated dataset with arrays that should have the same shapes as yours:
```python
import numpy as np

# 2000 stars, 10 times
flux2 = np.random.rand(2000, 10)
errors = np.random.rand(2000, 10)
time = np.linspace(0, 2000)
```

With that, I cannot reproduce the exception and the following code seems to work:
```python
from prose.fluxes import Fluxes

raw_fluxes = Fluxes(fluxes=flux2,time=time,errors=errors)
raw_fluxes.target = 1000
nan_stars = np.any(np.isnan(raw_fluxes.fluxes), axis=(0, 2)) # stars with nan fluxes
fluxes = raw_fluxes.mask_stars(~nan_stars)
diff_fluxes = raw_fluxes.autodiff()
```

Can you verify that your arrays are correctly shaped? You can also send files if that helps

Can you also check the version of prose you are working with? (`prose.__version__`)
